zrml/market-commons/src/migrations.rs

```diff
@@ -16,60 +16,481 @@
+                OldMarketStatus::Disputed => MarketStatus::Disputed,
+                OldMarketStatus::Resolved => MarketStatus::Resolved,
+            };
+            let new_market = Market {
+                base_asset: old_market.base_asset,
+                creator: old_market.creator,
+                creation: old_market.creation,
+                creator_fee: old_market.creator_fee,
+                oracle: old_market.oracle,
+                metadata: old_market.metadata,
+                market_type: old_market.market_type,
+                period: old_market.period,
+                deadlines: old_market.deadlines,
+                scoring_rule,
+                status,
+                report: old_market.report,
+                resolved_outcome: old_market.resolved_outcome,
+                dispute_mechanism: old_market.dispute_mechanism,
+                bonds: old_market.bonds,
+                early_close: old_market.early_close,
+            };
+            Some(new_market)
+        });
+        log::info!("MigrateScoringRuleAndMarketStatus: Upgraded {} markets.", translated);
+        total_weight =
+            total_weight.saturating_add(T::DbWeight::get().reads_writes(translated, translated));
+
+        StorageVersion::new(MARKET_COMMONS_NEXT_STORAGE_VERSION).put::<MarketCommons<T>>();
+        total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        log::info!("MigrateScoringRuleAndMarketStatus: Done!");
+        total_weight
     }
 
-    #[allow(unused)]
-    const SWAPS: &[u8] = b"Swaps";
-    #[allow(unused)]
-    const POOLS: &[u8] = b"Pools";
-    #[allow(unused)]
-    fn storage_prefix_of_swaps_pallet() -> [u8; 32] {
-        storage_prefix(b"Swaps", b":__STORAGE_VERSION__:")
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+        let old_markets = storage_key_iter::<MarketIdOf<T>, OldMarketOf<T>, Blake2_128Concat>(
+            MARKET_COMMONS,
+            MARKETS,
+        )
+        .collect::<BTreeMap<_, _>>();
+        let markets = Markets::<T>::iter_keys().count();
+        let decodable_markets = Markets::<T>::iter_values().count();
+        if markets == decodable_markets {
+            log::info!("All {} markets could successfully be decoded.", markets);
+        } else {
+            log::error!(
+                "Can only decode {} of {} markets - others will be dropped.",
+                decodable_markets,
+                markets
+            );
+        }
+
+        Ok(old_markets.encode())
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade(previous_state: Vec<u8>) -> Result<(), &'static str> {
+        let old_markets: BTreeMap<MarketId, OldMarketOf<T>> =
+            Decode::decode(&mut &previous_state[..]).unwrap();
+        let old_market_count = old_markets.len();
+        let new_market_count = Markets::<T>::iter().count();
+        assert_eq!(old_market_count, new_market_count);
```

Comment:
I assume since the tests below check correctness of transferred data and here it is ensured that the market count stays the same, we can assume that all markets got correctly migrated.
The only situation that I could imagine that causes a corrupted state is when some market data cannot be decoded. This already should be caught in `pre_upgrade`.